pkg/rsstorage/servers/file/server_test.go

```diff
@@ -712,6 +714,62 @@ func (s *FileStorageServerSuite) TestUsageTimeout(c *check.C) {
 	c.Assert(fsUsage.UsedBytes, check.Equals, 0*datasize.KB)
 }
 
+func (s *FileStorageServerSuite) TestDiskUsage(c *check.C) {
+	testFiles := 10
+	testStr := []byte("hello world")
+
+	expectedSize := 0
+
+	for i := 0; i < testFiles; i++ {
+		f, _ := os.CreateTemp("testdata", "*")
+		f.Write(testStr)
+		expectedSize += len(testStr)
+
+		defer os.Remove(f.Name())
+	}
+
+	sz, err := diskUsage("testdata", time.Minute, time.Minute)
+	c.Assert(err, check.IsNil)
+	c.Check(uint64(sz), check.Equals, uint64(expectedSize))
+}
+
+func (s *FileStorageServerSuite) TestDiskUsageLoop(c *check.C) {
```

Comment:
@jonyoder I read that creating a symlink would recreate the loop issue, but I wasn't able to reproduce it here. Any ideas?

Reply:
I'm not familiar with the issue, so I'm afraid not. Was this related to a bug that is documented elsewhere?

Reply:
Oh, maybe this was an older Go bug, I see something [for Windows ](https://github.com/golang/go/issues/17540) but it looks like it's been closed for a while.

I'm still not sure exactly what the customers are experiencing here, all I know is that this area of the code ends up hanging forever preventing RSPM boot up.